scripts/hudson/quickstart.sh

```diff
@@ -140,15 +140,19 @@ function configure_wildfly {
   cd $WORKSPACE
   wget --user=guest --password=guest -nv https://ci.wildfly.org/httpAuth/repository/downloadAll/WF_Nightly/.lastSuccessful/artifacts.zip
   unzip -q artifacts.zip
-  export WILDFLY_DIST_ZIP=$(ls wildfly-*Final-SNAPSHOT.zip | head -n 1)
-  [ "x$WILDFLY_DIST_ZIP" = "x" ] && echo "Cannot find WildFly distribution zip file" && return 1
-  unzip -q $WILDFLY_DIST_ZIP
-  export JBOSS_HOME=`pwd`/${WILDFLY_DIST_ZIP%.zip}
+  # the artifacts.zip may be wrapping several zip files: artifacts.zip -> wildfly-latest-SNAPSHOT.zip -> wildfly-###-SNAPSHOT.zip
+  local wildflyLatestZipWrapper=$(ls wildfly-latest-*.zip | head -n 1)
+  unzip -q $wildflyLatestZipWrapper
+  rm -f $wildflyLatestZipWrapper
+  local wildflyDistZip=$(ls wildfly-*-SNAPSHOT.zip | head -n 1)
+  [ "x$wildflyDistZip" = "x" ] && echo "Cannot find WildFly distribution zip file" && return 1
+  unzip -q ${wildflyDistZip}
+  export JBOSS_HOME=`pwd`/${wildflyDistZip%.zip}
```

Comment:
If there is no file in the zip matching the zip name minus the extension then this will fail (it only works because there is a wildfly policy to name the zip after the directory). A simple test ([ -d $JBOSS_HOME ]) would detect this case (in addition to validating that the zip is sane).

Reply:
+1, adding the test